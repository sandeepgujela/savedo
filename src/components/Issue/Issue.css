.issue-container{
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  border:1px solid #e0e4e8;
  border-top:none;
  padding:10px;
  cursor: pointer;
}

.issue-container:hover {
  background: #f6f8fa;
}

.link {
  text-decoration: none;
}

.icon-wrapper {
  padding: 0 8px;
  color:#28a745;
}

.title-wrapper {
  display: flex;
  flex-direction: column;
  width:92%
}

.title-wrapper .title{
  font-weight: 600;
  font-size:16px;
}

.title-wrapper .title:hover{
  color: #0366d6;
  cursor:pointer;
}

.title-wrapper .meta-wrapper{
  display: flex;
  flex-direction: row;
  color: #818284;
  font-size:12px;
  margin-top: 5px
}


.comments {
  width: 5%;
  color:black;
  font-size:14px;
  display: flex;
  justify-content: flex-end
}

.label-component{
  border-radius: 2px;
  box-shadow: inset 0 -1px 0 rgba(27,31,35,.12);
  font-size: 12px;
  font-weight: 600;
  height: 20px;
  line-height: 15px;
  padding: .15em 5px;
  margin-left: 5px;
}

@media only screen and (max-width: 620px) {

  .issue-container{
    padding:  3% 3% 3% 1%  ;
  }
  .icon-wrapper {
    padding: 0 4px;
  }
  .title-wrapper {
    width:95% ;
  }
  .title-wrapper .title{
    font-weight: 600;
    color:#0366d6;
    font-size:14px
  }
  .comments {
    display: none;
  }

}
