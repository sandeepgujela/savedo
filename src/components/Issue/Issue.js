import React, { Component } from 'react';
import PropTypes from 'prop-types';
import './Issue.css';
import Octicon, { IssueOpened, Comment } from '@githubprimer/octicons-react';
import { meaningFullTime } from '../../util/helpers';
class Issue extends Component {
  getLabels(labels) {
    if (!labels) return;
    return labels.map((label, index) => {
      const style = { backgroundColor: '#' + label.color };
      return (
        <span key={'label-' + index} className="label-component" style={style}>
          {label.name}
        </span>
      );
    });
  }
  render() {
    const {
      number,
      labels,
      title,
      created_at,
      user,
      comments,
    } = this.props.issue;
    return (
      <li className="issue-container">
        <div className="icon-wrapper">
          <Octicon icon={IssueOpened} size="small" verticalAlign="middle" />
        </div>

        <div className="title-wrapper">
          <div className="title">
            {title}
            {this.getLabels(labels)}
          </div>

          <div className="meta-wrapper">
            <div className="issue-number">#{number} </div>
            <div className="ml-5">Opened on {meaningFullTime(created_at)}</div>
            <div className="ml-5">By {user.login} </div>
          </div>
        </div>

        <div className="comments">
          {comments > 0 ? (
            <span>
              <Octicon icon={Comment} size="small" verticalAlign="middle" />{' '}
              {comments}
            </span>
          ) : null}
        </div>
      </li>
    );
  }
}

Issue.propTypes = {
  issue: PropTypes.shape({
    title: PropTypes.string.isRequired,
    number: PropTypes.number.isRequired,
    created_at: PropTypes.string.isRequired,
    html_url: PropTypes.string.isRequired,
    state: PropTypes.string.isRequired,
    labels: PropTypes.arrayOf(
      PropTypes.shape({
        color: PropTypes.string,
        name: PropTypes.string,
      }),
    ).isRequired,
    comments: PropTypes.number.isRequired,
  }).isRequired,
};

export default Issue;
