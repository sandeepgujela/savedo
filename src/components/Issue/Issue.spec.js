import React from 'react';
import { shallow } from 'enzyme';
import Issue from './Issue';

describe('<Issue />', () => {
  const build = passedProps => {
    const defaultProps = {
      issue: {
        html_url: 'https://github.com/facebook/react/issues/14213',
        id: 380109743,
        number: 14213,
        title:
          'Textarea loses focus after inserting paired punctuation with Chinese IME',
        user: {
          login: 'LastShadowNeo',
          id: 44996396,
        },
        labels: [
          {
            id: 127893911,
            name: 'Component: DOM',
            color: 'fef2c0',
          },
          {
            id: 40929151,
            name: 'Type: Bug',
            color: 'b60205',
          },
        ],
        state: 'open',
        comments: 2,
        created_at: '2018-11-13T08:26:59Z',
      },
    };
    return shallow(<Issue {...defaultProps} {...passedProps} />);
  };

  it('should render', () => {
    const component = build();
    expect(component).toMatchSnapshot();
  });
});
