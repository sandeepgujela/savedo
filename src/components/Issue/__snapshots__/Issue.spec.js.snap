// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issue /> should render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Issue
    issue={
      Object {
        "comments": 2,
        "created_at": "2018-11-13T08:26:59Z",
        "html_url": "https://github.com/facebook/react/issues/14213",
        "id": 380109743,
        "labels": Array [
          Object {
            "color": "fef2c0",
            "id": 127893911,
            "name": "Component: DOM",
          },
          Object {
            "color": "b60205",
            "id": 40929151,
            "name": "Type: Bug",
          },
        ],
        "number": 14213,
        "state": "open",
        "title": "Textarea loses focus after inserting paired punctuation with Chinese IME",
        "user": Object {
          "id": 44996396,
          "login": "LastShadowNeo",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="icon-wrapper"
        >
          <Octicon
            className="octicon"
            icon={[Function]}
            size="small"
            verticalAlign="middle"
          />
        </div>,
        <div
          className="title-wrapper"
        >
          <div
            className="title"
          >
            Textarea loses focus after inserting paired punctuation with Chinese IME
            <span
              className="label-component"
              style={
                Object {
                  "backgroundColor": "#fef2c0",
                }
              }
            >
              Component: DOM
            </span>
            <span
              className="label-component"
              style={
                Object {
                  "backgroundColor": "#b60205",
                }
              }
            >
              Type: Bug
            </span>
          </div>
          <div
            className="meta-wrapper"
          >
            <div
              className="issue-number"
            >
              #
              14213
               
            </div>
            <div
              className="ml-5"
            >
              Opened on 
              5 days ago
            </div>
            <div
              className="ml-5"
            >
              By 
              LastShadowNeo
               
            </div>
          </div>
        </div>,
        <div
          className="comments"
        >
          <span>
            <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />
             
            2
          </span>
        </div>,
      ],
      "className": "issue-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Octicon
            className="octicon"
            icon={[Function]}
            size="small"
            verticalAlign="middle"
          />,
          "className": "icon-wrapper",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "octicon",
            "icon": [Function],
            "size": "small",
            "verticalAlign": "middle",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title"
            >
              Textarea loses focus after inserting paired punctuation with Chinese IME
              <span
                className="label-component"
                style={
                  Object {
                    "backgroundColor": "#fef2c0",
                  }
                }
              >
                Component: DOM
              </span>
              <span
                className="label-component"
                style={
                  Object {
                    "backgroundColor": "#b60205",
                  }
                }
              >
                Type: Bug
              </span>
            </div>,
            <div
              className="meta-wrapper"
            >
              <div
                className="issue-number"
              >
                #
                14213
                 
              </div>
              <div
                className="ml-5"
              >
                Opened on 
                5 days ago
              </div>
              <div
                className="ml-5"
              >
                By 
                LastShadowNeo
                 
              </div>
            </div>,
          ],
          "className": "title-wrapper",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Textarea loses focus after inserting paired punctuation with Chinese IME",
                Array [
                  <span
                    className="label-component"
                    style={
                      Object {
                        "backgroundColor": "#fef2c0",
                      }
                    }
                  >
                    Component: DOM
                  </span>,
                  <span
                    className="label-component"
                    style={
                      Object {
                        "backgroundColor": "#b60205",
                      }
                    }
                  >
                    Type: Bug
                  </span>,
                ],
              ],
              "className": "title",
            },
            "ref": null,
            "rendered": Array [
              "Textarea loses focus after inserting paired punctuation with Chinese IME",
              Object {
                "instance": null,
                "key": "label-0",
                "nodeType": "host",
                "props": Object {
                  "children": "Component: DOM",
                  "className": "label-component",
                  "style": Object {
                    "backgroundColor": "#fef2c0",
                  },
                },
                "ref": null,
                "rendered": "Component: DOM",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": "label-1",
                "nodeType": "host",
                "props": Object {
                  "children": "Type: Bug",
                  "className": "label-component",
                  "style": Object {
                    "backgroundColor": "#b60205",
                  },
                },
                "ref": null,
                "rendered": "Type: Bug",
                "type": "span",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="issue-number"
                >
                  #
                  14213
                   
                </div>,
                <div
                  className="ml-5"
                >
                  Opened on 
                  5 days ago
                </div>,
                <div
                  className="ml-5"
                >
                  By 
                  LastShadowNeo
                   
                </div>,
              ],
              "className": "meta-wrapper",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "#",
                    14213,
                    " ",
                  ],
                  "className": "issue-number",
                },
                "ref": null,
                "rendered": Array [
                  "#",
                  14213,
                  " ",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Opened on ",
                    "5 days ago",
                  ],
                  "className": "ml-5",
                },
                "ref": null,
                "rendered": Array [
                  "Opened on ",
                  "5 days ago",
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "By ",
                    "LastShadowNeo",
                    " ",
                  ],
                  "className": "ml-5",
                },
                "ref": null,
                "rendered": Array [
                  "By ",
                  "LastShadowNeo",
                  " ",
                ],
                "type": "div",
              },
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <span>
            <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />
             
            2
          </span>,
          "className": "comments",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />,
              " ",
              2,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "className": "octicon",
                "icon": [Function],
                "size": "small",
                "verticalAlign": "middle",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            " ",
            2,
          ],
          "type": "span",
        },
        "type": "div",
      },
    ],
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="icon-wrapper"
          >
            <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />
          </div>,
          <div
            className="title-wrapper"
          >
            <div
              className="title"
            >
              Textarea loses focus after inserting paired punctuation with Chinese IME
              <span
                className="label-component"
                style={
                  Object {
                    "backgroundColor": "#fef2c0",
                  }
                }
              >
                Component: DOM
              </span>
              <span
                className="label-component"
                style={
                  Object {
                    "backgroundColor": "#b60205",
                  }
                }
              >
                Type: Bug
              </span>
            </div>
            <div
              className="meta-wrapper"
            >
              <div
                className="issue-number"
              >
                #
                14213
                 
              </div>
              <div
                className="ml-5"
              >
                Opened on 
                5 days ago
              </div>
              <div
                className="ml-5"
              >
                By 
                LastShadowNeo
                 
              </div>
            </div>
          </div>,
          <div
            className="comments"
          >
            <span>
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
               
              2
            </span>
          </div>,
        ],
        "className": "issue-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Octicon
              className="octicon"
              icon={[Function]}
              size="small"
              verticalAlign="middle"
            />,
            "className": "icon-wrapper",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "className": "octicon",
              "icon": [Function],
              "size": "small",
              "verticalAlign": "middle",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title"
              >
                Textarea loses focus after inserting paired punctuation with Chinese IME
                <span
                  className="label-component"
                  style={
                    Object {
                      "backgroundColor": "#fef2c0",
                    }
                  }
                >
                  Component: DOM
                </span>
                <span
                  className="label-component"
                  style={
                    Object {
                      "backgroundColor": "#b60205",
                    }
                  }
                >
                  Type: Bug
                </span>
              </div>,
              <div
                className="meta-wrapper"
              >
                <div
                  className="issue-number"
                >
                  #
                  14213
                   
                </div>
                <div
                  className="ml-5"
                >
                  Opened on 
                  5 days ago
                </div>
                <div
                  className="ml-5"
                >
                  By 
                  LastShadowNeo
                   
                </div>
              </div>,
            ],
            "className": "title-wrapper",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Textarea loses focus after inserting paired punctuation with Chinese IME",
                  Array [
                    <span
                      className="label-component"
                      style={
                        Object {
                          "backgroundColor": "#fef2c0",
                        }
                      }
                    >
                      Component: DOM
                    </span>,
                    <span
                      className="label-component"
                      style={
                        Object {
                          "backgroundColor": "#b60205",
                        }
                      }
                    >
                      Type: Bug
                    </span>,
                  ],
                ],
                "className": "title",
              },
              "ref": null,
              "rendered": Array [
                "Textarea loses focus after inserting paired punctuation with Chinese IME",
                Object {
                  "instance": null,
                  "key": "label-0",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Component: DOM",
                    "className": "label-component",
                    "style": Object {
                      "backgroundColor": "#fef2c0",
                    },
                  },
                  "ref": null,
                  "rendered": "Component: DOM",
                  "type": "span",
                },
                Object {
                  "instance": null,
                  "key": "label-1",
                  "nodeType": "host",
                  "props": Object {
                    "children": "Type: Bug",
                    "className": "label-component",
                    "style": Object {
                      "backgroundColor": "#b60205",
                    },
                  },
                  "ref": null,
                  "rendered": "Type: Bug",
                  "type": "span",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="issue-number"
                  >
                    #
                    14213
                     
                  </div>,
                  <div
                    className="ml-5"
                  >
                    Opened on 
                    5 days ago
                  </div>,
                  <div
                    className="ml-5"
                  >
                    By 
                    LastShadowNeo
                     
                  </div>,
                ],
                "className": "meta-wrapper",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "#",
                      14213,
                      " ",
                    ],
                    "className": "issue-number",
                  },
                  "ref": null,
                  "rendered": Array [
                    "#",
                    14213,
                    " ",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "Opened on ",
                      "5 days ago",
                    ],
                    "className": "ml-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    "Opened on ",
                    "5 days ago",
                  ],
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      "By ",
                      "LastShadowNeo",
                      " ",
                    ],
                    "className": "ml-5",
                  },
                  "ref": null,
                  "rendered": Array [
                    "By ",
                    "LastShadowNeo",
                    " ",
                  ],
                  "type": "div",
                },
              ],
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <span>
              <Octicon
                className="octicon"
                icon={[Function]}
                size="small"
                verticalAlign="middle"
              />
               
              2
            </span>,
            "className": "comments",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <Octicon
                  className="octicon"
                  icon={[Function]}
                  size="small"
                  verticalAlign="middle"
                />,
                " ",
                2,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "className": "octicon",
                  "icon": [Function],
                  "size": "small",
                  "verticalAlign": "middle",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
              2,
            ],
            "type": "span",
          },
          "type": "div",
        },
      ],
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
